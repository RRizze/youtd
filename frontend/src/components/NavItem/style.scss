.nav-item {
  color: #cadce8;
  text-transform: capitalize;
  font-weight: bold;
  transition: color .3s;
  &:hover {
    color: #ffffff;
  }
}