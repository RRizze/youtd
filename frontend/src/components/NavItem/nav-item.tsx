import { Link } from "react-router-dom";
import React, { ReactNode } from "react";
import "./style.scss";

interface Props {
  children: ReactNode;
  to: string;
}

const NavItem = ({ children, to }: Props) => (
  <Link
    to={to}
    className="nav-item"
  >
    {children}
  </Link>
);

export default NavItem;
