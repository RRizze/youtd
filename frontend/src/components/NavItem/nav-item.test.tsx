import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, mount } from "enzyme";
import NavItem from "./nav-item";

describe("NavItem tests", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router><NavItem children="About" to="/about" /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("render with props", () => {
    const props = {
      children: "about",
      to: "/about",
      className: "nav-item",
    },
      wrapper = shallow(<NavItem {...props} />);

    expect(wrapper.props()).toEqual(props);
  });

  describe("check prop to", () => {
    const props = {
      children: "about",
      to: "/about",
    },
      wrapper = shallow(<NavItem {...props} />);

    it("type for to is a string", () => {
      expect(typeof wrapper.props().to).toBe("string");
    });
  });
});
