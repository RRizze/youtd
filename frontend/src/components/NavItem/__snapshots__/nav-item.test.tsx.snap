// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem tests renders correctly 1`] = `
<a
  className="nav-item"
  href="/about"
  onClick={[Function]}
>
  About
</a>
`;
